client/components/Message/StatusIcon/index.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, memo, ReactElement } from 'react';
+import { Box, Icon } from '@rocket.chat/fuselage';
+
+import { IMessage } from '../../../../definition/IMessage';
+import Edited from './Edited';
+import Translated from './Translated';
+import Pinned from './Pinned';
+import Starred from './Starred';
+import SentByMail from './SentByMail';
+
+type StatusIconType = FC<{
+	type: string;
+	msg: IMessage;
+}>;
+
+const renderStatus = (type: string, msg: IMessage): ReactElement | undefined => {
```

Comment:
I'm not so sure about this pattern, since you're returning JSX, this is basically a component. Also, I think you should create a type for `type`, containing all possible types.

Reply:
@tassoevan @ggazzo, is it okay to have functions like these?

Reply:
No problem, I create a specific type for type prop...